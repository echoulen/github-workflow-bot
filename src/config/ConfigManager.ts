import {Config} from "@type/Config";

export class ConfigManager {
  public static getConfig(): Config {
    return {
      integration: process.env.integration as any,
      repository: {
        url: process.env.repoUrl,
        token: process.env.token,
      },
      notify: {
        enable: process.env.enableNotify === "true",
        token: process.env.notifyToken,
      },
      lgtm: {
        enable: process.env.enableLGTM === "true",
        count: Number(process.env.lgtmCount),
      },
    };
  }
}
