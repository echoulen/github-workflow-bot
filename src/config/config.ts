import {Config} from "../types/Config";

export const config: Config = {
  integration: process.env.integration as any,
  repository: {
    url: process.env.repoUrl,
    token: process.env.token,
  },
  notify: {
    enable: process.env.enableNotify === "true",
    token: process.env.notifyToken,
  },
  lgtm: {
    enable: process.env.enableLGTM === "true",
    count: Number(process.env.lgtmCount),
  },
};
