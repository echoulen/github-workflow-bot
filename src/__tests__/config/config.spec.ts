import {config} from "@config/config";

describe("config spec", () => {
  it("should get config proper", () => {
    expect(config).toMatchSnapshot();
  });
});
