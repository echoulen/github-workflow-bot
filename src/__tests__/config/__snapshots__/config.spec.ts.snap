// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config spec should get config proper 1`] = `
Object {
  "integration": "gitlab",
  "lgtm": Object {
    "count": 2,
    "enable": true,
  },
  "notify": Object {
    "enable": true,
    "token": false,
  },
  "repository": Object {
    "token": "gitlab-test-token",
    "url": "http://gitlab.com",
  },
}
`;
